File: Data Structures & Algorithms/three-integer-sum/submission-2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans = set()
        n = len(nums)
        nums = sorted(nums)

        for i in range(n):
            l, r = i + 1, n - 1
            target = -nums[i]
            while l < r:
                current_sum = nums[l] + nums[r]

                if current_sum == target:
                    ans.add((nums[i], nums[l], nums[r]))
                    l += 1
                    r -= 1
                elif current_sum < target:
                    l += 1
                else:
                    r -= 1
        return list(ans)

File: Data Structures & Algorithms/sqrtx/submission-2.cpp
class Solution {
public:
    int mySqrt(int x) {
        int l = 0, r = x;
        long mid = 0;
        int ans = 0;
        while (l <= r) {
            mid = (long) (l + r) / 2;
            if (mid * mid > x) {
                r = mid - 1;
                ans = r;
            } else if (mid * mid < x) {
                l = mid + 1;
            } else {
                return mid;
            }
        }
        return ans;
    }
};